viz_scripts/generic_metrics.ipynb

```diff
@@ -119,193 +120,396 @@
   },
   {
    "cell_type": "markdown",
-   "id": "modified-skiing",
+   "id": "6d44c87e",
    "metadata": {},
    "source": [
-    "## Generic Metrics"
+    "## Collect Data from Database for Sensed Metrics"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "55ec383b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "expanded_ct_sensed, file_suffix_sensed, quality_text_sensed, debug_df_sensed = scaffolding.load_viz_notebook_sensor_inference_data(year,\n",
+    "                                                                            month,\n",
+    "                                                                            program,\n",
+    "                                                                            include_test_users,\n",
+    "                                                                            sensed_algo_prefix)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8ecfd343",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a combined debug df from Generic and Sensed Metrics\n",
+    "merged_debug_df = debug_df.combine_first(debug_df_sensed)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1af1f02b",
+   "metadata": {},
+   "source": [
+    "## 100% Stacked Bar Charts"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "distributed-peace",
+   "id": "e50959c1",
    "metadata": {},
    "source": [
-    "### Distribution of Mode_confirm attribute"
+    "### 1. Represents Number of Trips"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "tracked-serbia",
-   "metadata": {
-    "scrolled": false
-   },
+   "id": "aef741fe",
+   "metadata": {},
    "outputs": [],
    "source": [
-    "file_name='ntrips_mode_confirm%s' % file_suffix\n",
     "plot_title_no_quality= \"Number of trips for each mode (selected by users)\"\n",
+    "file_name = f'ntrips_total{file_suffix}'\n",
+    "\n",
     "try:\n",
-    "    labels_mc = expanded_ct['Mode_confirm'].value_counts(dropna=True).keys().tolist()\n",
-    "    values_mc = expanded_ct['Mode_confirm'].value_counts(dropna=True).tolist()    \n",
-    "    plot_title = plot_title_no_quality+\"\\n\"+quality_text\n",
-    "    pie_chart_mode(plot_title,labels_mc,values_mc,colors_mode,file_name)\n",
-    "    alt_text = store_alt_text_pie(pd.DataFrame(values_mc, labels_mc), file_name, plot_title)\n",
-    "    print(expanded_ct['Mode_confirm'].value_counts(dropna=True))\n",
+    "    df_confirmed_tc = pd.DataFrame()\n",
+    "    df_sensed_tc = pd.DataFrame()\n",
```

Comment:
This is initialized because in cases when `expanded_ct` and/or `expanded_ct_sensed` data frame doesn't any data in it. `df_confirmed_tc` and `df_sensed_tc` would not be formed. These empty data frames are initialized to handle and represent the bars with the message as `"No data available for ..."`